File: Test2/Options/UserOptionsForm.vb
Imports System.Windows.Forms
Imports Inventor

Namespace Options
    Public Class UserOptionsForm
        Private _options As New UserOptions
        Public Sub New()
            InitializeComponent()
            _options = UserOptions.Load()
        End Sub

        Private Sub UserOptionsForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
            _options = UserOptions.Load()
            PEXLoc.Text = _options.PrintExportLocation
            DXFexLoc.Text = _options.DxfExportLocation

            ' Apply theme colors
            ApplyThemeColors()
        End Sub

        Private Sub ApplyThemeColors()
            Dim oThemeManager As ThemeManager
            oThemeManager = ThisApplication.ThemeManager
            Dim oTheme As Theme
            oTheme = oThemeManager.ActiveTheme
            If oTheme.Name = "LightTheme" Then
                ' Dark theme colors
                FutureMsg.BackColor = Drawing.Color.FromArgb(CByte(245), CByte(245), CByte(245))
                FutureMsg.ForeColor = Drawing.Color.Black
                BtnSave.FlatAppearance.BorderColor = Drawing.Color.FromArgb(CByte(186), CByte(186), CByte(186))
                BtnSave.FlatAppearance.MouseOverBackColor = Drawing.Color.White
                PEXLoc.BackColor = Drawing.Color.White
                PEXLoc.ForeColor = Drawing.Color.Black
                PrintExportLocationButton.FlatAppearance.BorderColor = Drawing.Color.FromArgb(CByte(186), CByte(186), CByte(186))
                PrintExportLocationButton.FlatAppearance.MouseOverBackColor = Drawing.Color.White
                PrintExportLocationButton.ForeColor = Drawing.Color.Black
                DXFexLoc.BackColor = Drawing.Color.White
                DXFexLoc.ForeColor = Drawing.Color.Black
                DXFExportLocationButton.FlatAppearance.BorderColor = Drawing.Color.FromArgb(CByte(186), CByte(186), CByte(186))
                DXFExportLocationButton.FlatAppearance.MouseOverBackColor = Drawing.Color.White
                BackColor = Drawing.Color.FromArgb(CByte(245), CByte(245), CByte(245))
                ForeColor = Drawing.Color.Black
                SCBackground.BackColor = Drawing.Color.FromArgb(CByte(245), CByte(245), CByte(245))
                BtnCncl.FlatAppearance.MouseOverBackColor = Drawing.Color.White
                BtnCncl.FlatAppearance.BorderColor = Drawing.Color.FromArgb(CByte(186), CByte(186), CByte(186))
            Else
            End If
        End Sub

        Private Sub BtnSave_Click(sender As Object, e As EventArgs) Handles BtnSave.Click
            _options.PrintExportLocation = PEXLoc.Text
            _options.DxfExportLocation = DXFexLoc.Text
            _options.Save()
            Close()
        End Sub

        Private Sub PrintExportLocationButton_Click(sender As Object, e As EventArgs) _
            Handles PrintExportLocationButton.Click
            Dim folderBrowser As New FolderBrowserDialog With {
                    .Description = "Select Print Export Location"
                    }
            folderBrowser.ShowDialog()
            PEXLoc.Text = folderBrowser.SelectedPath
        End Sub

        Private Sub DXFExportLocationButton_Click(sender As Object, e As EventArgs) _
            Handles DXFExportLocationButton.Click
            Dim folderBrowser As New FolderBrowserDialog With {
                    .Description = "Select DXF Export Location"
                    }
            folderBrowser.ShowDialog()
            ' Assuming you have a text box to display the selected path
            DXFexLoc.Text = folderBrowser.SelectedPath
        End Sub
        Private Sub BtnCncl_Click(sender As Object, e As EventArgs) Handles BtnCncl.Click
            Close()
        End Sub
    End Class
End Namespace

File: Doyle Addin C/Options/UserOptionsForm.cs
using System;
using Color = System.Drawing.Color;
using System.Windows.Forms;
using Inventor;

namespace Doyle_Addin.Options
{
    public partial class UserOptionsForm
    {
        private UserOptions options = new UserOptions();

        public UserOptionsForm()
        {
            this.InitializeComponent();
            options = UserOptions.Load();
        }

        private void UserOptionsForm_Load(object sender, EventArgs e)
        {
            options = UserOptions.Load();
            this.PEXLoc.Text = options.PrintExportLocation;
            this.DXFexLoc.Text = options.DxfExportLocation;
            this.ChkObsoletePrint.Checked = options.EnableObsoletePrint;

            // Apply theme colors
            ApplyThemeColors();
        }

        private void ApplyThemeColors()
        {
            ThemeManager oThemeManager;
            oThemeManager = Doyle_Addin.GlobalsHelpers.ThisApplication.ThemeManager;
            Theme oTheme;
            oTheme = oThemeManager.ActiveTheme;
            if (oTheme.Name == "LightTheme")
            {
                // Dark theme colors
                this.FutureMsg.BackColor = Color.FromArgb(245, 245, 245);
                this.FutureMsg.ForeColor = Color.Black;
                this.BtnSave.FlatAppearance.BorderColor = Color.FromArgb(186, 186, 186);
                this.BtnSave.FlatAppearance.MouseOverBackColor = Color.White;
                this.PEXLoc.BackColor = Color.White;
                this.PEXLoc.ForeColor = Color.Black;
                this.PrintExportLocationButton.FlatAppearance.BorderColor = Color.FromArgb(186, 186, 186);
                this.PrintExportLocationButton.FlatAppearance.MouseOverBackColor = Color.White;
                this.PrintExportLocationButton.ForeColor = Color.Black;
                this.DXFexLoc.BackColor = Color.White;
                this.DXFexLoc.ForeColor = Color.Black;
                this.DXFExportLocationButton.FlatAppearance.BorderColor = Color.FromArgb(186, 186, 186);
                this.DXFExportLocationButton.FlatAppearance.MouseOverBackColor = Color.White;
                BackColor = Color.FromArgb(245, 245, 245);
                ForeColor = Color.Black;
                this.SCBackground.BackColor = Color.FromArgb(245, 245, 245);
                this.BtnCncl.FlatAppearance.MouseOverBackColor = Color.White;
                this.BtnCncl.FlatAppearance.BorderColor = Color.FromArgb(186, 186, 186);
                this.FeaturesPanel.BackColor = Color.FromArgb(245, 245, 245);
                this.ChkObsoletePrint.BackColor = Color.FromArgb(245, 245, 245);
                this.ChkObsoletePrint.ForeColor = Color.Black;
            }
            else
            {
                // Dark theme colors (keep existing dark theme as default)
                this.FeaturesPanel.BackColor = Color.FromArgb(59, 68, 83);
                this.ChkObsoletePrint.BackColor = Color.FromArgb(59, 68, 83);
                this.ChkObsoletePrint.ForeColor = Color.White;
            }
        }

        private void BtnSave_Click(object sender, EventArgs e)
        {
            options.PrintExportLocation = this.PEXLoc.Text;
            options.DxfExportLocation = this.DXFexLoc.Text;
            options.EnableObsoletePrint = this.ChkObsoletePrint.Checked;
            options.Save();
            DialogResult = DialogResult.OK;
            Close();
        }

        private void PrintExportLocationButton_Click(object sender, EventArgs e)
        {
            var folderBrowser = new FolderBrowserDialog() { Description = "Select Print Export Location" };
            folderBrowser.ShowDialog();
            this.PEXLoc.Text = folderBrowser.SelectedPath;
        }

        private void DXFExportLocationButton_Click(object sender, EventArgs e)
        {
            var folderBrowser = new FolderBrowserDialog() { Description = "Select DXF Export Location" };
            folderBrowser.ShowDialog();
            // Assuming you have a text box to display the selected path
            this.DXFexLoc.Text = folderBrowser.SelectedPath;
        }

        private void BtnCncl_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }
    }
}